flytekit/remote/execution.py

```diff
@@ -0,0 +1,128 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List, Union
+
+from flytekit import ContainerTask
+from flytekit.annotated import context_manager as flyte_context
+from flytekit.annotated.context_manager import SerializationSettings
+from flytekit.annotated.promise import translate_inputs_to_literals
+from flytekit.clients.friendly import SynchronousFlyteClient
+from flytekit.common.core.identifier import Identifier, WorkflowExecutionIdentifier
+from flytekit.common.exceptions import user as user_exceptions
+from flytekit.common.translator import get_serializable
+from flytekit.common.types.helpers import get_sdk_type_from_literal_type, pack_python_std_map_to_literal_map
+from flytekit.common.workflow_execution import SdkWorkflowExecution
+from flytekit.configuration import auth as auth_config
+from flytekit.configuration import platform as platform_config
+from flytekit.models.common import Annotations, AuthRole, Labels, Notification
+from flytekit.models.core.identifier import ResourceType
+from flytekit.models.execution import ExecutionMetadata, ExecutionSpec, NotificationList
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskSpec
+
+
+@dataclass
+class ExecutionParameters(object):
+    notifications: List[Notification] = None
+    labels: Labels = None
+    annotations: Annotations = None
+    auth_role: AuthRole = None
+    # optional name to assign to execution
+    name: str = None
+    # The name of the user launching this execution.
+    principal: str = None
+
+
+def execute(
+    entity: Union[Identifier, ContainerTask],
+    project: str,
+    domain: str,
+    inputs: Dict[str, Any] = None,
```

Comment:
should we inputs as a dictionary or just kwargs? Whats your gut feeling?

Reply:
kwargs gives us consistency with just calling the function in local execute mode but b/c there are required/positional args in this scenario naming inputs isn't what departs from that pattern. what do you think? happy to change or keep as is